### PR TITLE
[ new ] Added an inverse of inject₁ in Data.Fin 

### PR DESCRIPTION
**Pull request by L-TChen**
Added the inverse of inject₁:

lower₁ : (i : Fin (suc n)) → (n ≢ toℕ i) → Fin n

and its related properties:

toℕ-inject₁-≢   : (i : Fin n)                 → n ≢ toℕ (inject₁ i)
inject₁-lower₁  : (ne : n ≢ toℕ i)         → lower₁ (reduce₁ i ne) ≡ i
lower₁-inject₁′ : (i : Fin n) (ne : n ≢ toℕ (inject₁ i)) → lower₁ (inject₁ i) ne ≡ i
lower₁-inject₁  : (i : Fin n) → lower₁ (inject₁ i) (toℕ-inject₁-≢ i) ≡ i

Please feel free to comment or suggest. These are general properties one may need while manipulating Fin's index using inject₁.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -485,12 +485,17 @@ Other minor additions
...
+  cast   : m ≡ n → Fin m → Fin n
+  lower₁ : (i : Fin (suc n)) → (n ≢ toℕ i) → Fin n
   ```
 
 * Added new proof to `Data.Fin.Properties`:
   ```agda
-  toℕ-cast    : toℕ (cast eq k) ≡ toℕ k
+  toℕ-cast        : toℕ (cast eq k) ≡ toℕ k
+  toℕ-inject₁-≢   : (i : Fin n)              → n ≢ toℕ (inject₁ i)
+  inject₁-lower₁  : (ne : n ≢ toℕ i)         → lower₁ (reduce₁ i ne) ≡ i
```
Comment (gallais) [inline CHANGELOG.md]: This should be `inject₁-lower₁ : (ne : n ≢ toℕ i) → inject₁ (lower₁ i ne) ≡ i`

File: src/Data/Fin/Properties.agda [hunk 2]
```diff
@@ -340,6 +340,30 @@ toℕ-inject₁ : ∀ {n} (i : Fin n) → toℕ (inject₁ i) ≡ toℕ i
...
+inject₁-lower₁ {suc n} (suc i) ne
+  rewrite inject₁-lower₁ i  (λ x → ne (cong suc x)) = refl
+
+lower₁-inject₁′ : ∀ {n} (i : Fin n) (ne : n ≢ toℕ (inject₁ i))
+                → lower₁ (inject₁ i) ne ≡ i
+lower₁-inject₁′ {zero} () ne
+lower₁-inject₁′ {suc n} zero ne = refl
+lower₁-inject₁′ {suc n} (suc i) ne
+  rewrite lower₁-inject₁′ i (λ x → ne (cong suc x)) = refl
+
```
Comment (gallais) [inline src/Data/Fin/Properties.agda]: Here pattern-matching on `i` is enough to force `{n}` to have the right shape. So you can write:

```agda
lower₁-inject₁′ : ∀ {n} (i : Fin n) (ne : n ≢ toℕ (inject₁ i))
                → lower₁ (inject₁ i) ne ≡ i
lower₁-inject₁′ zero    ne = refl
lower₁-inject₁′ (suc i) ne = cong suc (lower₁-inject₁′ i (ne ∘ cong suc))
```

File: src/Data/Fin/Base.agda [hunk 3]
```diff
@@ -122,6 +122,14 @@ inject≤ : ∀ {m n} → Fin m → m ℕ.≤ n → Fin n
 inject≤ zero    (s≤s le) = zero
 inject≤ (suc i) (s≤s le) = suc (inject≤ i le)
 
+-- lower₁ "n" _ = "n".
```
Comment (MatthewDaggitt) [inline src/Data/Fin/Base.agda]: Should this be `lower₁ "i" _ = "i"` instead?

**Comment by MatthewDaggitt** [comment]
Looks good. Thanks for the PR!

File: src/Data/Fin/Base.agda (hunk 3 above)
Comment (L-TChen) [reply src/Data/Fin/Base.agda]: I am actually puzzled by the comment in this module. 

For example, shouldn't the following comment
```agda
-- raise m "n" = "m + n".
raise : ∀ {m} n → Fin m → Fin (n ℕ.+ m)
raise zero    i = i
raise (suc n) i = suc (raise n i)
```
be 
```
-- raise m "n" = "m + n".

Comment (MatthewDaggitt) [reply src/Data/Fin/Base.agda]: Yup! Good spot. It'd be great if you could change that as well.

Comment (L-TChen) [reply src/Data/Fin/Base.agda]: Great. I will go through this module later.

Comment (L-TChen) [reply src/Data/Fin/Base.agda]: They should be renamed accordingly now.

**Comment by gallais** [comment]
@L-TChen If you have time, it'd be nice to have a proof that in `lower₁`,
the proof `n ≢ toℕ i` does not matter. That is to say:

```agda
lower₁-ext : (i : Fin (suc n)) (p q : n ≢ toℕ i) → lower₁ i p ≡ lower₁ i q
```

(not sure what a good name for it is)

**Comment by L-TChen** [comment]
> @L-TChen If you have time, it'd be nice to have a proof that in `lower₁`,
> the proof `n ≢ toℕ i` does not matter. That is to say:
> 
> ```agda
> lower₁-ext : (i : Fin (suc n)) (p q : n ≢ toℕ i) → lower₁ i p ≡ lower₁ i q
> ```
> 
> (not sure what a good name for it is)

I am also not sure what the name should be ...but here it is.

**Comment by MatthewDaggitt** [comment]
Thanks, will rename it `lower₁-irrelevant`.

**Pull request merged**